I improved the fix a bit and forced push the changes. You can easily test it with
```
docker build -t gr-gsm-3.10 -f tests/dockerfiles/Ubuntu_22_04.docker  .
```
One test is still failing, I'm working on fixing it.
```
Running tests...
Test project /src/build
      Start  1: qa_arfcn
 1/19 Test  #1: qa_arfcn ...........................   Passed    0.43 sec
      Start  2: qa_decryption
 2/19 Test  #2: qa_decryption ......................   Passed    0.42 sec
      Start  3: qa_burst_printer
 3/19 Test  #3: qa_burst_printer ...................   Passed    0.42 sec
      Start  4: qa_message_printer
 4/19 Test  #4: qa_message_printer .................   Passed    0.43 sec
      Start  5: qa_burst_file_source
 5/19 Test  #5: qa_burst_file_source ...............***Failed    1.90 sec
/usr/lib/python3/dist-packages/gnuradio/gr/top_block.py:48: DeprecationWarning: setDaemon() is deprecated, set the daemon attribute instead
  self.setDaemon(1)
Fterminate reached from thread id: 7faab4b0b640Got pmt::exception
pmt::deserialize: malformed input stream, tag value = : 244
0x7faadd3be5d9: (gr::tagged_stream_block::check_topology(int, int)+0x2e49)
0x7faadd03928c: (std::rethrow_exception(std::__exception_ptr::exception_ptr)+0x7c)
0x7faadd0392f7: (std::terminate()+0x17)
0x7faadd039558: (__cxa_throw+0x48)
0x7faadd2d34e6: (pmt::read(std::istream&)+0xd1a)
0x7faadd2f73a7: (pmt::serialize_str[abi:cxx11](std::shared_ptr<pmt::pmt_base>)+0x3d7)
0x7faadd2f7b10: (pmt::deserialize(std::basic_streambuf<char, std::char_traits<char> >&)+0x460)
0x7faab6bf4455: (gr::gsm::burst_file_source_impl::run()+0x1e5)
0x7faadcc540cb: (boost::detail::make_ready_at_thread_exit(boost::shared_ptr<boost::detail::shared_state_base>)+0x2fb)
0x7faadd9f5b43: (pthread_condattr_setpshared+0x513)
0x7faadda86bb4: (clone+0x44)
Aborted (core dumped)

      Start  6: qa_burst_timeslot_splitter
 6/19 Test  #6: qa_burst_timeslot_splitter .........   Passed    0.53 sec
      Start  7: qa_burst_sdcch_subslot_splitter
 7/19 Test  #7: qa_burst_sdcch_subslot_splitter ....   Passed    0.49 sec
      Start  8: qa_burst_timeslot_filter
 8/19 Test  #8: qa_burst_timeslot_filter ...........   Passed    0.45 sec
      Start  9: qa_burst_sdcch_subslot_filter
 9/19 Test  #9: qa_burst_sdcch_subslot_filter ......   Passed    0.42 sec
      Start 10: qa_burst_fnr_filter
10/19 Test #10: qa_burst_fnr_filter ................   Passed    0.43 sec
      Start 11: qa_dummy_burst_filter
11/19 Test #11: qa_dummy_burst_filter ..............   Passed    0.43 sec
      Start 12: qa_tch_f_decoder
12/19 Test #12: qa_tch_f_decoder ...................   Passed    0.51 sec
      Start 13: qa_tch_h_decoder
13/19 Test #13: qa_tch_h_decoder ...................   Passed    0.45 sec
      Start 14: qa_tch_f_chans_demapper
14/19 Test #14: qa_tch_f_chans_demapper ............   Passed    0.48 sec
      Start 15: qa_tch_h_chans_demapper
15/19 Test #15: qa_tch_h_chans_demapper ............   Passed    0.55 sec
      Start 16: qa_gsm_bcch_ccch_demapper
16/19 Test #16: qa_gsm_bcch_ccch_demapper ..........   Passed    0.44 sec
      Start 17: qa_gsm_bcch_ccch_sdcch4_demapper
17/19 Test #17: qa_gsm_bcch_ccch_sdcch4_demapper ...   Passed    0.45 sec
      Start 18: qa_gsm_sdcch8_demapper
18/19 Test #18: qa_gsm_sdcch8_demapper .............   Passed    0.42 sec
      Start 19: qa_txtime_setter
19/19 Test #19: qa_txtime_setter ...................   Passed    0.41 sec

95% tests passed, 1 tests failed out of 19

Total Test time (real) =  10.07 sec

The following tests FAILED:
	  5 - qa_burst_file_source (Failed)
Errors while running CTest
make: *** [Makefile:71: test] Error 8
The command '/bin/sh -c cmake .. &&     mkdir $HOME/.grc_gnuradio/ $HOME/.gnuradio/ &&     make -j $(nproc) &&     make install &&     ldconfig &&     make CTEST_OUTPUT_ON_FAILURE=1 test' returned a non-zero code: 2
```

--- a/tests/dockerfiles/Debian_testing.docker
+++ b/tests/dockerfiles/Debian_testing.docker
@@ -1,5 +1,5 @@
-FROM debian:testing
-MAINTAINER Piotr Krysik
+FROM fedora:36
+MAINTAINER Vasil Velichkov
 
 RUN apt-get update && DEBIAN_FRONTEND=noninteractive apt-get install -y \
     cmake \

--- a/tests/dockerfiles/Fedora_36.Dockerfile
+++ b/tests/dockerfiles/Fedora_36.Dockerfile
@@ -1,0 +1,48 @@
+FROM fedora:36
+
+RUN dnf install -y \
+        gcc-c++ \
+        make \
+        cmake \
+        pkgconfig \
+        boost-devel \
+        spdlog-devel \
+        gmp-devel \
+        pybind11-devel \
+        gnuradio-devel \
+        libosmocore-devel \
+        gr-osmosdr \
+        swig \
+        doxygen \
+        python3-docutils \
+        cppunit-devel \
+#deps of libosmocore \
+        autoconf \
+        automake \
+        libtool \
+        gnutls-devel \
+        libusb-devel \
+        libmnl-devel \
+        lksctp-tools-devel
+
+ADD https://gitea.osmocom.org/osmocom/libosmocore/archive/1.7.0.tar.gz /src/libosmocore.tar.gz
+RUN cd /src/ && \
+        tar -xzvf libosmocore.tar.gz && \
+        cd libosmocore && \
+        autoreconf -if && \
+        ./configure --disable-pcsc && \
+        make && \
+        make install
+
+COPY ./ /src/gr-gsm
+WORKDIR /src/gr-gsm/build
+
+RUN cmake .. && \
+        # The parallel build sometimes fails when the .grc_gnuradio
+        # and .gnuradio directories do not exist
+        mkdir $HOME/.grc_gnuradio/ $HOME/.gnuradio/ && \
+        make && \
+        make -j $(nproc) && \
+        make install && \
+        ldconfig && \
+        make CTEST_OUTPUT_ON_FAILURE=1 test

